storage_service/locations/models/package.py

```diff
@@ -2014,6 +2017,14 @@ def delete_from_storage(self):
             )
         self.status = self.DELETED
         self.save()
+
+        # Remove size from location and space used values. Like the initial
+        # checks and updates, this is not considering race conditions.
+        space.used -= self.size
+        space.save()
+        location.used -= self.size
+        location.save()
```

Comment:
Not sure how often we do this in practice, but should these three updates be part of a single transaction?

```python
with transaction.atomic():
    ...
```

Reply:
Discussed this offline. Radda reminded me that we're going to be addressing this issue and similar issues in Storage Service as part of the async refactoring.